database/sinkdb/sinkdb.go

```diff
@@ -0,0 +1,132 @@
+// Package sinkdb provides a strongly consistent key-value store.
+package sinkdb
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/golang/protobuf/proto"
+
+	"chain/database/raft"
+	"chain/database/sinkdb/internal/sinkpb"
+)
+
+// Open initializes the key-value store and returns a database handle.
+func Open(laddr, dir, bootURL string, httpClient *http.Client, useTLS bool) (*DB, error) {
+	state := newState()
+	sv, err := raft.Start(laddr, dir, bootURL, httpClient, useTLS, state)
+	if err != nil {
+		return nil, err
+	}
+	db := &DB{state: state, raft: sv}
+	return db, nil
+}
+
+// DB provides access to an opened kv store.
+type DB struct {
+	state *state
+	raft  *raft.Service
+}
+
+type Op struct {
+	err     error
+	conds   []*sinkpb.Cond
+	effects []*sinkpb.Op
+}
+
+// IfNotExists encodes a conditional to make an instruction
+// successful only if the provided key does not exist.
+func IfNotExists(key string) (op Op) {
+	op.conds = append(op.conds, &sinkpb.Cond{
+		Type: sinkpb.Cond_NOT_KEY_EXISTS,
+		Key:  key,
+	})
+	return op
+}
+
+// Delete encodes a delete operation for key.
+func Delete(key string) (op Op) {
```

Comment:
Nit: these return signatures should prob just be `Op` (rather than `(op Op)` which is stutter and doesn't add any useful info).

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Yeah, even if the body of the method is a little uglier (like one extra `var op Op` line or whatev) the better interface outweighs that imo.